### ISSUE TITLE
Not wokr rule no-unmodified-loop-condition

### ISSUE THREAD

**Issue by boggddan** [eslint/eslint#8817]
Tell us about your environment

ESLint Version:
v4.0.0
Node Version:
v8.0.0
npm Version:
v5.0.0
Hi, maybe I finded bug.
Not wokr rule 'no-unmodified-loop-condition'

```js
/* OFF another errors */
/*eslint indent: "off", no-var: "off", no-undef: "off", quotes: "off", "no-unused-vars": "off", no-console: "off", no-redeclare: 'off', no-implicit-globals: 'off', no-constant-condition: "off", no-duplicate-class-member: 'off', no-magic-numbers: 'off' */

/*eslint no-unmodified-loop-condition: "error" */


while (node) {
    doSomething(node);
}
node = other;

for (var j = 0; j < items.length; ++i) {
    doSomething(items[j]);
}

while (node !== root) {
    doSomething(node);
}
```
bash and VS Code editor show only one error, on the last variable `root`, and not show error with variable `node` and 'j` 

```bash
$ eslint src/js/test3.js

D:\ProjectRuby\table_test\src\js\test3.js
  16:17  error  'root' is not modified in this loop  no-unmodified-loop-condition
```

EsLint Demo does not show errors!
![2017-06-27_13h01_19](https://user-images.githubusercontent.com/24915025/27582271-fdf6f0c0-5b38-11e7-8daf-ea4b37bc43d8.png)

**Comment by mysticatea** [comment]
> EsLint Demo does not show errors!

I think it because the `node` variable is not declared.

**Comment by boggddan** [comment]
@mysticatea, Thanks. I took samples from documentation on this rule [no-unmodified-loop-condition](http://eslint.org/docs/rules/no-unmodified-loop-condition).
Could you give me a working samples?

**Comment by Aladdin-ADD** [comment]
@boggddan working sample:
```js
for (var j = 0; j < 10;) {}
```
I guess it is related to `j < items.length;`, if change `items.length` to Literal(such as 10), it is working.

**Comment by platinumazure** [comment]
Possible duplicate of #8788.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
